llm_studio/src/datasets/text_causal_language_modeling_ds.py

```diff
@@ -413,10 +387,9 @@ def _read_data(self, idx: int, sample: Dict) -> Dict:
             )
         )
 
-        if not self.cfg.training.use_rlhf or self.mode != "train":
-            samples[-1][1] = torch.empty(0)
-
-        prompt_input_ids = torch.cat([torch.cat(sample) for sample in samples])
+        # Remove last answer from encoding to create the prompt for training
```

Comment:
for "inference"
it's used in the generate function and can probably be further refactored there